src/compute-types/src/dataflows.rs

```diff
@@ -726,106 +726,82 @@ impl Arbitrary for DataflowDescription<OptimizedMirRelationExpr, (), mz_repr::Ti
-            any::<(GlobalId, ComputeSinkDesc<(), mz_repr::Timestamp>)>(),
-            1..3,
-        ),
-        as_of_some in any::<bool>(),
-        as_of in proptest::collection::vec(any::<mz_repr::Timestamp>(), 1..5),
-        debug_name in ".*",
-        initial_storage_as_of_some in any::<bool>(),
-        initial_as_of in proptest::collection::vec(any::<mz_repr::Timestamp>(), 1..5),
-        refresh_schedule_some in any::<bool>(),
-        refresh_schedule in any::<RefreshSchedule>(),
-        time_dependence in any::<Option<TimeDependence>>(),
-    ) -> DataflowDescription<OptimizedMirRelationExpr, (), mz_repr::Timestamp> {
-        DataflowDescription {
-            source_imports: BTreeMap::from_iter(source_imports.into_iter()),
-            index_imports: BTreeMap::from_iter(index_imports.into_iter()),
-            objects_to_build,
-            index_exports: BTreeMap::from_iter(index_exports.into_iter()),
-            sink_exports: BTreeMap::from_iter(
-                sink_descs.into_iter(),
-            ),
-            as_of: if as_of_some {
-                Some(Antichain::from(as_of))
-            } else {
-                None
-            },
-            until: Antichain::new(),
-            initial_storage_as_of: if initial_storage_as_of_some {
-                Some(Antichain::from(initial_as_of))
-            } else {
-                None
-            },
-            refresh_schedule: if refresh_schedule_some {
-                Some(refresh_schedule)
-            } else {
-                None
+fn any_dataflow_description<P, S, T>(
+    any_source_import: impl Strategy<Value = (GlobalId, (SourceInstanceDesc<S>, bool))>,
+) -> impl Strategy<Value = DataflowDescription<P, S, T>>
+where
+    P: Arbitrary,
+    S: 'static + Arbitrary,
+    T: Arbitrary + timely::PartialOrder,
+    ComputeSinkDesc<S, T>: Arbitrary,
+{
+    (
+        proptest::collection::vec(any_source_import, 1..3),
+        proptest::collection::vec(any_dataflow_index_import(), 1..3),
+        proptest::collection::vec(any::<BuildDesc<P>>(), 1..3),
+        proptest::collection::vec(any_dataflow_index_export(), 1..3),
+        proptest::collection::vec(any::<(GlobalId, ComputeSinkDesc<S, T>)>(), 1..3),
+        any::<bool>(),
+        proptest::collection::vec(any::<T>(), 1..5),
+        any::<bool>(),
+        proptest::collection::vec(any::<T>(), 1..5),
+        any::<bool>(),
+        any::<RefreshSchedule>(),
+        any::<Option<TimeDependence>>(),
+    )
+        .prop_map(
+            |(
+                source_imports,
+                index_imports,
+                objects_to_build,
+                index_exports,
+                sink_descs,
+                as_of_some,
+                as_of,
+                initial_storage_as_of_some,
+                initial_as_of,
+                refresh_schedule_some,
+                refresh_schedule,
+                time_dependence,
+            )| DataflowDescription {
```

Comment:
This should fix the issue you're seeing below. It's because the [`Strategy`](https://docs.rs/proptest/latest/proptest/strategy/trait.Strategy.html#method.prop_map) trait, and thus `prop_map` is only implemented for tuples of 12 elements and less. The debug name is the 13th element so the tuple would no longer implement `Strategy`. Nesting the tuples will fix the issue

```suggestion
    (
        (
            proptest::collection::vec(any_source_import, 1..3),
            proptest::collection::vec(any_dataflow_index_import(), 1..3),
            proptest::collection::vec(any::<BuildDesc<P>>(), 1..3),
            proptest::collection::vec(any_dataflow_index_export(), 1..3),
            proptest::collection::vec(any::<(GlobalId, ComputeSinkDesc<S, T>)>(), 1..3),
            any::<bool>(),
            proptest::collection::vec(any::<T>(), 1..5),
            any::<bool>(),
            proptest::collection::vec(any::<T>(), 1..5),
            any::<bool>(),
            any::<RefreshSchedule>(),
            any::<Option<TimeDependence>>(),
        ),
        (
            any::<String>(),
        )
    )
        .prop_map(
            |(
                (
                    source_imports,
                    index_imports,
                    objects_to_build,
                    index_exports,
                    sink_descs,
                    as_of_some,
                    as_of,
                    initial_storage_as_of_some,
                    initial_as_of,
                    refresh_schedule_some,
                    refresh_schedule,
                    time_dependence,
                ),
                (
                    debug_name,
                )
            )| DataflowDescription {
```